### PR TITLE
Feat/mint components changes

### PR DESCRIPTION
**Pull request by aditiSoniWeb**
## What does this PR do?

New Mint component changes - 
Button
Checkbox
RadioButton
Loader
MessageBox
Chips -> Pills
IconButton
ToggleSwitch

## Types of changes

What types of changes does your code introduce to this project?

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist before merging

_Put an `x` in the boxes that apply_

- [ ] These changes have been thoroughly tested.

- [ ] Changes need to be immediately published on npm.

### CODE REVIEW

File: packages/ui-toolkit/src/components/atoms/CheckBox/CheckBox.tsx [hunk 1]
```diff
@@ -45,73 +52,78 @@ const CheckBox = (props: Props) => {
...
-  addParentClass: '',
-  activeColor: 'var(--purple500)',
-  inActiveColor: 'var(--gray900)',
+  isDisabled: false,
   labelComponent: () => null,
-  checkBoxDirection: CHECKBOX_DIRECTION.LEFT,
+  checkBoxDirection: 'Left',
   dataTestId: ''
-} as DefaultProps;
+} as unknown as DefaultProps;
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/CheckBox/CheckBox.tsx:126]: unknown? is this needed?

File: packages/ui-toolkit/src/components/atoms/CheckBox/checkBox.css [hunk 2]
```diff
@@ -1,21 +1,30 @@
-.c11AlignCenter {
-  display: flex;
-  align-items: center;
+.c11Default {
+width: fit-content;
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/CheckBox/checkBox.css:2]: formatting

File: packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { Loader, LOADER_TYPE } from '../Loader';
+
+import './iconButton.css';
+import { ReactIconProps } from '@groww-tech/icon-store';
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx:7]: sort the imports

File: packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      'ib31IsLoading': isLoading
+    });
+
+  return (
+    isLoading
+      ? <Loader loaderType={LOADER_TYPE.CIRCULAR}
+        loaderClassName='iconLoader'
+      />
+      : <div className={iconClasses}>
+        {iconComponent(iconProps as any)}
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx:37]: remove any

File: packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css [hunk 5]
```diff
@@ -0,0 +1,36 @@
+.ib31Icon {
+width: 24px;
+height: 24px;
+}
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css:4]: formatting

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    bodySmall: size === 'Small' || size === 'XSmall',
+    bodyBase: size === 'Base',
+    bodyLarge: size === 'Large',
+    bodyXLarge: size === 'XLarge'
+  });
+
+  return (
+    <div
+      className={
+        cn(
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx:42]: take this in a variable

File: packages/ui-toolkit/src/components/atoms/Pill/pill.css [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  gap: 8px;
+  border-radius: 99px;
+}
+
+.pill12Outlined {
+  border: 1px solid var(--gray150);
+}
+
+.pill12SelectedAccentPill {
+ background-color: var(--green100);
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/Pill/pill.css:13]: can we use tokens?

File: packages/ui-toolkit/src/components/atoms/CheckBox/CheckBox.tsx (hunk 1 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/CheckBox/CheckBox.tsx:126]: removed

File: packages/ui-toolkit/src/components/atoms/CheckBox/checkBox.css (hunk 2 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/CheckBox/checkBox.css:2]: fixed

File: packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx (hunk 3 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx:7]: fixed

File: packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx (hunk 4 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/IconButton/IconButton.tsx:37]: replaced with ReactIconProp

File: packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css (hunk 5 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css:4]: fixed

File: packages/ui-toolkit/src/components/atoms/Button/button.css [hunk 8]
```diff
@@ -1,116 +1,164 @@
...
-  color: var(--white);
+.btn96SecondaryButtonWithoutAccent:hover {
+  background-color: var(--gray50);
 }
 
-.btn51Secondary {
-  border: 1px solid var(--green500);
-  color: var(--green500);
+.btn96SecondaryButtonWithoutAccentDisabled {
+  background-color: var(--gray150);
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/Button/button.css:70]: token?

File: packages/ui-toolkit/src/components/atoms/Button/button.css [hunk 9]
```diff
@@ -1,116 +1,164 @@
...
+  border-right-color: transparent !important; 
+}
+
+.btn96MediumButton .btn96LoaderSize {
+  margin-top: 4px;
+  width: 20px; 
+  height: 20px;
+}
+
+.btn96LargeButton .btn96LoaderSize {
```
Comment (shubhamgavali1998) [inline packages/ui-toolkit/src/components/atoms/Button/button.css:138]:
```
.btn96LoaderSize {
     margin-top: 4px;
}
```
This is repeated. Can take this out and write it once

File: packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+ border: 2px solid var(--gray900);
+ padding: 4px;
+}
+
+.ib31IconDefault:hover {
+ background-color: var(--gray100);
+}
+
+.ib31Disabled {
+ background-color: var(--gray50);
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css:21]: token

File: packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+.ib31Disabled .ib31Icon {
+  color: var(--gray400);
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css:25]: token

File: packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+.iconLoader {
+ width: 22px;
+ height: 22px;
+ border: 2px solid var(--gray900);
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css:31]: token

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 13]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { Info } from '@groww-tech/icon-store/mi';
+
+import './messageBox.css';
+
+const BACKGROUNDS = {
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:8]: use from constants

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+  const {
+    size,
+    isIconPresent,
+    background,
+    content,
+    dataTestId,
+    isCompact
+  } = props;
+
+  const parentDivClass = cn({
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:26]: rename baseClasses

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+    'valign-wrapper infbd45ParentDiv': true,
+    'backgroundTertiary': background === BACKGROUNDS.NEUTRAL,
+    'backgroundWarningSubtle': background === BACKGROUNDS.WARNING,
+    'backgroundNegativeSubtle': background === BACKGROUNDS.ERROR,
+    'backgroundPositiveSubtle': background === BACKGROUNDS.POSITIVE,
+    'mb76CompactBox': isCompact === true
+  });
+
+  const labelClassNames = cn({
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:35]: labelClasses

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+      <div className='valign-wrapper'>
+        {
+          isIconPresent &&
+        <Info
+          size={20}
+          color='contentPrimary'
+        />
+        }
+      </div>
+      <span className={cn('contentPrimary', labelClassNames)}>{content}</span>
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:56]: contentPrimary can be in labelClasses

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+  const labelClassNames = cn({
+    'bodySmall': size === 'Small' || size === 'XSmall',
+    'bodyBase': size === 'Base',
+    'bodyLarge': size === 'Large',
+    'bodyXLarge': size === 'XLarge'
+  });
+
+  return (
+    <div
+      className={cn(parentDivClass, isCompact ? `mb45${size}` : 'mb76CompactBox')}
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:44]: get everything in baseClasses

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+
+type RequiredProps = {
+  content: string;
+};
+
+
+type DefaultProps = {
+  isIconPresent: boolean;
+  dataTestId: string;
+  size: 'XSmall' | 'Small' | 'Base' | 'Large' | 'XLarge';
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:70]: use from constants

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+  background: 'Neutral' | 'Error' | 'Warning' | 'Positive';
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:71]: use from constants

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+  isCompact: boolean;
+};
+
+
+MessageBox.defaultProps = {
+  isIconPresent: true,
+  background: 'Neutral',
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:78]: use from constants

File: packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+  dataTestId: '',
+  isOutlined: false,
+  size: 'XSmall',
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/MessageBox/MessageBox.tsx:81]: use from constants

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+
+    onClick(e);
+  };
+
+  const pillIconProps = {
+    size: 20
+  };
+
+  const labelClassName = cn({
+    bodySmall: size === 'Small' || size === 'XSmall',
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx:33]: use from constants

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+    trailingIcon,
+    isAccent,
+    isSelected,
+    isOutlined
+  } = props;
+
+
+  const handleClick = (e: React.MouseEvent<HTMLImageElement>) => {
+    e.stopPropagation();
+    const { onClick } = props;
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx:23]: can be destructured in the above block

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+            pill12SelectedPill: !isAccent && isSelected,
+            'backgroundPositiveSubtle contentAccent borderAccent': isSelected && isAccent
+          }
+        )
+      }
+      onClick={handleClick}
+    >
+      <span className='valign-wrapper'>{leadingIcon?.(pillIconProps)}</span>
+
+      <span className={labelClassName}>{text}</span>
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx:63]: labelClasses

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+
+type RequiredProps = {
+  text: string;
+}
+
+
+type DefaultProps = {
+  size: 'XSmall' | 'Small' | 'Base' | 'Large' | 'XLarge';
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx:77]: use from constants

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx [hunk 26]
```diff
@@ -0,0 +1,98 @@
...
+  isAccent: boolean;
+  isOutlined: boolean;
+  leadingIcon: ((props: ReactIconProps) => JSX.Element) | null;
+  trailingIcon: ((props: ReactIconProps) => JSX.Element) | null;
+};
+
+export type Props = RequiredProps & DefaultProps;
+
+Pill.defaultProps = {
+  size: 'Base',
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx:89]: use from constants

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+          }
+        )
+      }
+      onClick={handleClick}
+    >
+      <span className='valign-wrapper'>{leadingIcon?.(pillIconProps)}</span>
+
+      <span className={labelClassName}>{text}</span>
+
+      <span>{trailingIcon?.(pillIconProps)}</span>
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx:65]: <span>{trailingIcon?.(pillIconProps)}</span>
 can be
 
  {trailingIcon && <span>{trailingIcon.(pillIconProps)}</span>}

File: packages/ui-toolkit/src/components/atoms/RadioButton/RadioButton.tsx [hunk 28]
```diff
@@ -6,82 +6,91 @@ import { RadioButtonChecked, RadioButtonUnchecked } from '@groww-tech/icon-store
 import './radioButton.css';
 
 export const RADIO_DIRECTION = {
-  LEFT: 'LEFT',
-  RIGHT: 'RIGHT'
+  LEFT: 'Left',
+  RIGHT: 'Right'
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/RadioButton/RadioButton.tsx:10]: use from constants

File: packages/ui-toolkit/src/components/atoms/RadioButton/RadioButton.tsx [hunk 29]
```diff
@@ -6,82 +6,91 @@ import { RadioButtonChecked, RadioButtonUnchecked } from '@groww-tech/icon-store
...
+    }
+  };
+
   return (
     <div
-      onClick={onSelect}
+      onClick={radioButtonClick}
       data-test-id={dataTestId.length ? dataTestId : null}
-      className={`radioCo11Box ${parentClassName} ${radioDirection === RADIO_DIRECTION.RIGHT ? 'radioCo11BoxReverse' : ''}`}
+      className={
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/RadioButton/RadioButton.tsx:47]: baseClasses ?

File: packages/ui-toolkit/src/components/molecules/RadioButtonGroup/radioButtonGroupTypes.ts [hunk 30]
```diff
@@ -3,5 +3,5 @@ export type RadioButtonType = {
   label: React.ReactNode;
   labelClassName?: string;
   parentClassName?: string;
-  radioDirection?: string;
+  radioDirection?: 'Left' | 'Right';
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/molecules/RadioButtonGroup/radioButtonGroupTypes.ts:6]: use from constants

**Review by AkshayNaikGroww — CHANGES_REQUESTED** [review]

File: packages/ui-toolkit/src/components/atoms/Button/button.css [hunk 31]
```diff
@@ -1,116 +1,164 @@
...
+  position: absolute;
+}
+
+.btn96PrimaryButtonLoader {
+  border: 2px solid var(--dangerouslySetPrimaryBg);
+  border-top-color: transparent;
+}
+
+.btn96LoaderWithoutAccent {
+  border: 2px solid var(--gray900) !important;
```
[the remark this answers is not in the document]
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Button/button.css:128]: this border is used for loader, if borderNeutral is used.. it is not applying the changes..
green loader is only visible
border: 2px solid var(--gray900)
Screenshot 2023-11-22 at 3 44 08 PM

File: packages/ui-toolkit/src/components/atoms/Button/button.css [hunk 32]
```diff
@@ -1,116 +1,164 @@
...
+}
+
+.btn96PrimaryButtonLoader {
+  border: 2px solid var(--dangerouslySetPrimaryBg);
+  border-top-color: transparent;
+}
+
+.btn96LoaderWithoutAccent {
+  border: 2px solid var(--gray900) !important; 
+  border-right-color: transparent !important;
```
[the remark this answers is not in the document]
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Button/button.css:129]: fixed

File: packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css [hunk 33]
```diff
@@ -0,0 +1,33 @@
...
+.ib31IconDefault {
+ width: fit-content;
+ height: fit-content;
+ border-radius: 50%;
+ padding: 6px;
+}
+
+
+.ib31IconSelected {
+ border: 2px solid var(--gray900);
```
[the remark this answers is not in the document]
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/IconButton/iconButton.css:12]: fixed

File: packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx (hunk 6 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Pill/Pill.tsx]: fixed

File: packages/ui-toolkit/src/components/atoms/Pill/pill.css (hunk 7 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Pill/pill.css]: fixed

File: packages/ui-toolkit/src/components/atoms/Button/button.css (hunk 8 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Button/button.css]: fixed

File: packages/ui-toolkit/src/components/atoms/Button/button.css (hunk 9 above)
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Button/button.css:122]: fixed

File: packages/ui-toolkit/src/components/atoms/Button/Button.tsx [hunk 34]
```diff
@@ -1,162 +1,222 @@
...
+
+const VARIANTS = {
+  PRIMARY: 'Primary',
+  SECONDARY: 'Secondary',
+  TERTIARY: 'Tertiary',
+  POSITIVE: 'Positive',
+  NEGATIVE: 'Negative'
+};
+
+const SIZES = {
```
[the remark this answers is not in the document]
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Button/Button.tsx]: fixed

File: packages/ui-toolkit/src/components/atoms/Button/Button.tsx [hunk 35]
```diff
@@ -1,162 +1,222 @@
...
+    btn96ButtonHover: !isDisabled
+  });
+
+  const fontClasses = cn({
+    'bodySmallHeavy': size === SIZES.SMALL,
+    'bodyBaseHeavy': size === SIZES.BASE,
+    'bodyLargeHeavy': size === SIZES.LARGE,
+    'bodyXLargeHeavy': size === SIZES.XLARGE,
+    'contentDisabled': isDisabled && !isAccent,
+    'borderPrimary': isDisabled && !isAccent && variant !== VARIANTS.TERTIARY,
```
[the remark this answers is not in the document]
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Button/Button.tsx]: fixed

File: packages/ui-toolkit/src/utils/constants/index.ts [hunk 36]
```diff
@@ -68,3 +68,22 @@ export const DARK_THM = {
...
+
+export const SIZES = {
+  XSMALL: 'XSmall',
+  SMALL: 'Small',
+  BASE: 'Base',
+  LARGE: 'Large',
+  XLARGE: 'XLarge'
+} as const;
+
+export const POSITION = {
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/utils/constants/index.ts:86]: ICON_POSITION

**Review by AkshayNaikGroww — CHANGES_REQUESTED** [review]

File: packages/ui-toolkit/src/components/atoms/Button/button.css [hunk 37]
```diff
@@ -1,116 +1,145 @@
...
+.btn96PrimaryButtonLoader {
+  border: 2px solid var(--dangerouslySetPrimaryBg); /* the loader color will always bewhite in dark as well as in light theme */
+  border-top-color: transparent;
 }
 
-.btn51Secondary {
-  border: 1px solid var(--green500);
-  color: var(--green500);
+.btn96LoaderWithoutAccent {
+  border: 2px solid var(--gray900);
```
[the remark this answers is not in the document]
Comment (aditiSoniWeb) [reply packages/ui-toolkit/src/components/atoms/Button/button.css:109]: priority issue, borderNeutral doesn't apply grey900

File: packages/ui-toolkit/src/components/atoms/Loader/loader.constants.ts [hunk 38]
```diff
@@ -0,0 +1,9 @@
+export const LOADER_SIZE = {
```
Comment (AkshayNaikGroww) [inline packages/ui-toolkit/src/components/atoms/Loader/loader.constants.ts:1]: import SIZES and spread it here

**Review by AkshayNaikGroww — CHANGES_REQUESTED** [review]

**Review by shubhamgavali1998 — APPROVED** [review]

**Review by AkshayNaikGroww — APPROVED** [review]

**Pull request merged**